beetsplug/edit.py

```diff
@@ -0,0 +1,443 @@
+        # value. The ID of an item will never be changed.
+        self.not_fields = self.config['not_fields'].as_str_seq()
+
+        # the separator in your config sets the separator that will be used
+        # between fields in your terminal. Defaults to -
+        self.sep = self.config['sep'].get(unicode)
+
+        self.ed = None
+        self.ed_args = None
+        self.brw = None
+        self.brw_args = None
+
+    def commands(self):
+        edit_command = Subcommand(
+            'edit',
+            help='interactively edit metadata')
+        edit_command.parser.add_option(
+            '-e', '--extra',
+            action='store',
+            help='add additional fields to edit',
+        )
+        edit_command.parser.add_option(
+            '--all',
+            action='store_true', dest='all',
+            help='edit all fields',
+        )
+        edit_command.parser.add_option(
+            '--sum',
+            action='store_true', dest='sum',
+            help='list fields with the same value',
+        )
+        edit_command.parser.add_all_common_options()
+        edit_command.func = self.editor_music
+        return[edit_command]
+
+    def editor_music(self, lib, opts, args):
+        if self.editor:
+            self.ed_args = self.editor[1:] if len(self.editor) > 1 else None
+            self.ed = self.editor[0] if self.editor else None
+        if self.browser:
+            self.brw_args = self.browser[1:] if len(self.browser) > 1 else None
+            self.brw = self.browser[0] if self.browser else None
+
+        # Get the objects to edit.
+        query = decargs(args)
+        items, albums = _do_query(lib, query, opts.album, False)
+        objs = albums if opts.album else items
+        if not objs:
+            print_('Nothing to edit.')
+            return
+
+        # edit tags in a textfile in yaml-style
+        # makes a string representation of an object
+        # for now yaml but we could add html,pprint,toml
+        self.print_items = {
+            'yaml': self.print_to_yaml}
+        # makes an object from a string representation
+        # for now yaml but we could add html,pprint,toml
+        self.string_to_dict = {
+            'yaml': self.yaml_to_dict}
+        # 4 ways to view the changes in objects
+        self.diffresults = {
+            'ndiff': self.ndiff,
+            'unified': self.unified,
+            'html': self.html,
+            'vimdiff': self.vimdiff}
+        # make a dictionary from the chosen fields
+        # you can do em all or a selection
+        self.make_dict = {
+            'all': self.get_all_fields,
+            "selected": self.get_selected_fields}
+
+        self.get_fields_from(objs, opts)
+
+        # Confirm.
+        for obj in objs:
+            print_(format(obj))
+        if not ui.input_yn(ui.colorize('action_default', "Edit? (y/n)"), True):
+            return
+
+        dict_from_objs = self.make_dict[self.pick](self.fields, objs, opts)
```

Comment:
Diego,
The decision-proces has already disappeared in the fog of wine and cigars. I think mainly if it worked it was alright. I think at a certain moment II looked at the output of the Yaml and then used that as a guidance but if you get it running with the more natural list of dict, why not. Might save the extra call to `same_format`. Remember I'm not a programmer, more of a tinker. But ask away, maybe I can shed some light.